blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
+            #get particle positions
+            particle_pos = []
+            for particle in basis_particles:
+                particle_pos.append(context_pos[particle])
+            new_coord = self._findNewCoord(particle_pos[0], particle_pos[1], particle_pos[2], com)
+            #keep this in for now to check code is correct
+            #old_coord should be equal to com
+            old_coord = self._findOldCoord(particle_pos[0], particle_pos[1], particle_pos[2], new_coord)
+            n_dartboard.append(new_coord)
+            dartboard.append(old_coord)
 
-        probabilities: list of floats, default=None
-            A list of n probabilities,
-            where probabilities[i] corresponds to the probaility of moves[i]
-            being selected to perform its associated move() method.
+        self.n_dartboard = n_dartboard
+        self.dartboard = dartboard
 
-            If None, uniform probabilities are assigned.
+    def move(self, nc_context):
+        """
+        Function for performing smart darting move with darts that
+        depend on particle positions in the system
         """
 
-    #make a list from moves if not a list
-        if isinstance(moves,list):
-            self.moves = moves
-        else:
-            self.moves = [moves]
-        #normalize probabilities
-        if probabilities is None:
-            single_prob = 1. / len(self.moves)
-            self.probs = [single_prob for x in (self.moves)]
-        else:
-            prob_sum = float(sum(probabilities))
-            self.probs = [x/prob_sum for x in probabilities]
-        #if move and probabilitiy lists are different lengths throw error
-        if len(self.moves) != len(self.probs):
-            print('moves and probability list lengths need to match')
-            raise IndexError
-
-
-    def runEngine(self, context):
-        """Selects a random Move object based on its
-        assigned probability and and performs its move() function
-        on a context.
-        Parameters
-        ----------
-        context : openmm.context object
-        OpenMM context whose positions should be moved.
-        """
-        rand_num = np.random.choice(len(self.probs), p=self.probs)
-        try:
-            new_context = self.moves[rand_num].move(context)
-        except Exception as e:
-            #In case the move isn't properly implemented, print out useful info
-            print('Error: move not implemented correctly, printing traceback:')
-            ex_type, ex, tb = sys.exc_info()
-            traceback.print_tb(tb)
-            print(e)
-            raise SystemExit
-
-        return new_context
+        atom_indices = self.atom_indices
+        context = nc_context
+
+        stateinfo = context.getState(True, True, False, True, True, False)
+        oldDartPos = stateinfo.getPositions(asNumpy=True)
+        lig_pos = np.asarray(oldDartPos._value)[self.atom_indices]*unit.nanometers
+        self._findDart(context)
```

Comment:
Can you add comments explaining the algorithm here? I THINK what you're doing is:
a0) Update dart positions
a) Find the ligand center of mass
b) Find the dart region you are in, if any, and the change vector to get to the center of that dart
c) Pick a random dart and find the vector that would translate you from the center of the current dart region to the center of that dart region, which constitutes a center-of-mass move
d) Take the current center-of-mass and translate it by that center-of-mass move

Is that right? It took me a while to figure this out, and would have been harder if I didn't remember roughly what we're trying to do here. The comments should just SAY what you are doing so I don't have to figure it out.